Tests/MessagingInApp/MessagingInAppIntegrationTests.swift

```diff
@@ -269,15 +311,23 @@ extension MessagingInAppIntegrationTest {
         return true
     }
 
+    /*
+     The goal of this function is to determine the message that is visible on the screen to the user.
+     To try and make the tests the most accurate, access the currently displayed message from the UI layer of the code instead of business logic.
+     */
     var currentlyShownModalMessage: Message? {
-        guard let messageManager = Gist.shared.getModalMessageManager() else {
-            return nil // no modal message shown or loading
+        // Unfortunately, this method of determining visible modals is not 100% reliable.
+        // We are accessing the visible ViewController by getting manager instances from the Gist class and the Gist class destroys and creates
+        // instances of managers.
+        // It would be better if we could talk directly to the UI layer to get the visible ViewController or Gist has only 1 modal manager like the Android SDK days.
```

Comment:
As stated in the PR description, this PR is one step to making our modal integration tests more reliable. This call-out is suggesting another step to make the tests more reliable. 

There are some tests that check if a modal has been dismissed. While the test pass, it is not truly testing is a modal is shown in the UI or not. 

I think a great follow-up PR would be a refactor to the Gist class which makes modal message manager a singleton, just like the Android SDK does today. This would allow us to check the state of the 1 modal message in the app, which would make our tests more reliable.